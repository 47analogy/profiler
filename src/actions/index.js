export function waitingForProfileFromAddon() {
  return {
    type: 'WAITING_FOR_PROFILE_FROM_ADDON'
  };
}

export function receiveProfileFromAddon(profile) {
  return {
    type: 'RECEIVE_PROFILE_FROM_ADDON',
    profile: profile
  };
}

export function requestingSymbolTable(requestedLib) {
  return {
    type: 'REQUESTING_SYMBOL_TABLE',
    requestedLib
  };
}

export function receivedSymbolTableReply(requestedLib) {
  return {
    type: 'RECEIVED_SYMBOL_TABLE_REPLY',
    requestedLib
  };
}

export function startSymbolicating() {
  return {
    type: 'START_SYMBOLICATING'
  };
}

export function profileSymbolicationStep(profile) {
  return {
    type: 'PROFILE_SYMBOLICATION_STEP',
    profile
  };
}

export function doneSymbolicating() {
  return { type: 'DONE_SYMBOLICATING' };
}

export function mergeFunctions(threadIndex, oldFuncToNewFuncMap) {
  return {
    type: 'MERGE_FUNCTIONS',
    threadIndex, oldFuncToNewFuncMap
  };
}

export function assignFunctionNames(threadIndex, funcIndices, funcNames) {
  return {
    type: 'ASSIGN_FUNCTION_NAMES',
    threadIndex, funcIndices, funcNames
  };
}

export function changeSelectedFuncStack(selectedFuncStack) {
  return {
    type: 'CHANGE_SELECTED_FUNC_STACK',
    selectedFuncStack
  };
}
