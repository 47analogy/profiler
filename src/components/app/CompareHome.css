.compareHome {
  width: 75%;
  margin: auto;
  padding: 8em;

  border: 1px solid #CCC;
  border-radius: 3px;
  box-shadow: 0 5px 25px #0b1f50;
  background: #fff;
}

.compareHomeForm {
  display: grid;
  grid-template-columns: auto 1fr;
  grid-gap: 1em;
  align-items: center;
}

.compareHomeSubmitButton {
  grid-column-start: span 2
}

.compareHomeFormLabel {
  white-space: nowrap;
}

@media (max-width: 450px) {
  .compareHome {
    width: auto;
    padding: 1em;
  }
}
