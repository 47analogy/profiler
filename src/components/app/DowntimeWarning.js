/* This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
// @flow

import React, { PureComponent } from 'react';

import { Warning } from '../shared/Warning';

const LOCAL_STORAGE_KEY = 'profileServerMigrationNoticeDismissed2020';

type State = {|
  wasDismissedInThePast: boolean,
|};

export class DowntimeWarning extends PureComponent<{||}, State> {
  state = {
    wasDismissedInThePast: !!localStorage.getItem(LOCAL_STORAGE_KEY),
  };

  _onDowntimeActionClick() {
    window.open('/docs/#/./cloud-migration', '_blank');
  }

  _onClose() {
    localStorage.setItem(LOCAL_STORAGE_KEY, 'dismissed');
  }

  render() {
    // Because this code is planned to be only temporary, let's skip it in tests.
    if (process.env.NODE_ENV === 'test') {
      return null;
    }

    const currentDate = new Date();
    const feb23Date = new Date(Date.UTC(2020, 1, 23));
    if (currentDate >= feb23Date) {
      return null;
    }

    const feb21Date = new Date(Date.UTC(2020, 1, 21));
    if (currentDate < feb21Date && this.state.wasDismissedInThePast) {
      return null;
    }

    return (
      <Warning
        message="The profiler will encounter some downtime on Feb 21 and 22."
        actionText="More information"
        actionTitle="Consult more information about the downtime (opens a new tab)"
        actionOnClick={this._onDowntimeActionClick}
        onClose={this._onClose}
      />
    );
  }
}
