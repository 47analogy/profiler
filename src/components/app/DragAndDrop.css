.dragAndDropArea {
  isolation: isolate;
}

.dragAndDropMessageWrapper {
  position: absolute;
  top: 0;
  left: 0;
  display: flex;
  width: 100%;
  height: 100%;
  box-sizing: border-box;
  align-items: stretch;
  justify-content: center;
  padding: 1em;
  background-color: rgba(255, 255, 255, 0.9);
  opacity: 0;
  pointer-events: none;
  transition: opacity 200ms;
}

.dragAndDropMessageWrapper.dragging {
  opacity: 1;
}

.dragAndDropMessage {
  display: flex;
  width: 100%;
  align-items: center;
  justify-content: center;
  border: 2px dashed #888;
  border-radius: 10px;
  color: #666;
  font-size: 30px;
  text-align: center;
}
