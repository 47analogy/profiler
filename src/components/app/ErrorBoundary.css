.appErrorBoundary {
  display: flex;
  overflow: hidden;
  box-sizing: border-box;
  flex: 1;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  padding: 10px;
  background-color: var(--grey-20);
}

.appErrorBoundaryContents {
  display: flex;
  width: 100%;
  max-width: 600px;
  max-height: calc(100%);
  box-sizing: border-box;
  flex-direction: column;
}

@keyframes appErrorBoundaryDetailsFadeIn {
  0% {
    opacity: 0;
  }
  100% {
    opacity: 1;
  }
}

.appErrorBoundaryDetails {
  flex-shrink: 1;
  padding: 25px;
  margin: 12px 0;
  animation-duration: 200ms;
  animation-name: appErrorBoundaryDetailsFadeIn;
  background: #fff;
  border-radius: 4px;
  box-shadow: 0 0 10px rgba(0, 0, 0, 0.1);
  color: var(--grey-70);
  font-family: monospace;
  line-height: 1.3;
  overflow-y: auto;
  white-space: pre-wrap;
}
