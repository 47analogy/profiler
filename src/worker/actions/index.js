import { summarizeProfile } from '../../common/summarize-profile';

export function processProfileSummary() {
  return function (dispatch, getState) {
    dispatch({
      toContent: true,
      type: 'PROFILE_SUMMARY_PROCESSED',
      summary: summarizeProfile(getState().profile),
    });
  };
}

export function profileProcessed(profile) {
  return {
    type: 'PROFILE_PROCESSED',
    profile: profile,
  };
}
