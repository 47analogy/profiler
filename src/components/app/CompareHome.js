/* This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at http://mozilla.org/MPL/2.0/. */

// @flow

import React, { PureComponent } from 'react';
import { changeProfilesToCompare } from '../../actions/app';

import explicitConnect from '../../utils/connect';
import type {
  ExplicitConnectOptions,
  ConnectedProps,
} from '../../utils/connect';

import './CompareHome.css';

type DispatchProps = {|
  +changeProfilesToCompare: typeof changeProfilesToCompare,
|};

type Props = ConnectedProps<{||}, {||}, DispatchProps>;

type State = {
  profile1: string,
  profile2: string,
};

class CompareHome extends PureComponent<Props, State> {
  state = { profile1: '', profile2: '' };

  handleInputChange = (event: SyntheticInputEvent<>) => {
    const { name, value } = event.target;
    this.setState({ [name]: value });
  };

  handleFormSubmit = (e: SyntheticEvent<>) => {
    e.preventDefault();
    let { profile1, profile2 } = this.state;
    const { changeProfilesToCompare } = this.props;
    if (!profile1.startsWith('http')) {
      profile1 = 'http://' + profile1;
    }
    if (!profile2.startsWith('http')) {
      profile2 = 'http://' + profile2;
    }
    changeProfilesToCompare(profile1, profile2);
  };

  render() {
    const { profile1, profile2 } = this.state;

    return (
      <form className="compareHome" onSubmit={this.handleFormSubmit}>
        <h1>Enter the 2 profile URLs that youâ€™d like to compare</h1>
        <p>
          The tool will extract the data from the selected track and range for
          each profile, and put them both on the same view to make them easy to
          compare.
        </p>
        <label htmlFor="compareHomeProfile1">Profile 1:</label>
        <input
          name="profile1"
          id="compareHomeProfile1"
          className="compareHome-input"
          onChange={this.handleInputChange}
          value={profile1}
        />
        <label htmlFor="compareHomeProfile2">Profile 2:</label>
        <input
          name="profile2"
          id="compareHomeProfile2"
          className="compareHome-input"
          onChange={this.handleInputChange}
          value={profile2}
        />
        <input
          className="compareHome-submit-button"
          type="submit"
          value="Retrieve profiles"
        />
      </form>
    );
  }
}

const options: ExplicitConnectOptions<{||}, {||}, Props> = {
  mapDispatchToProps: { changeProfilesToCompare },
  component: CompareHome,
};

export default explicitConnect(options);
