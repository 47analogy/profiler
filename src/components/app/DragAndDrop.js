/* This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at http://mozilla.org/MPL/2.0/. */

// @flow

import * as React from 'react';
import classNames from 'classnames';
import { retrieveProfileFromFile } from '../../actions/receive-profile';
import type { ConnectedProps } from '../../utils/connect';

import './DragAndDrop.css';

function _dragPreventDefault(event: DragEvent) {
  event.preventDefault();
}

type OwnProps = {|
  +className?: string,
  +children?: React.Node,
  +render?: React.Node => React.Node,
|};

type DispatchProps = {|
  +retrieveProfileFromFile: typeof retrieveProfileFromFile,
|};

type DragAndDropState = {
  isDragging: boolean,
};

type DragAndDropProps = ConnectedProps<OwnProps, {||}, DispatchProps>;

// TODO Add documentation

/**
 * Creates a target area to drop files on. A dropped file will be
 * loaded into the profiler.
 */
class DragAndDrop extends React.PureComponent<
  DragAndDropProps,
  DragAndDropState
> {
  state = {
    isDragging: false,
  };

  componentDidMount() {
    // Prevent dropping files on the document.
    document.addEventListener('drag', _dragPreventDefault, false);
    document.addEventListener('dragover', _dragPreventDefault, false);
    document.addEventListener('drop', _dragPreventDefault, false);
  }

  componentWillUnmount() {
    document.removeEventListener('drag', _dragPreventDefault, false);
    document.removeEventListener('dragover', _dragPreventDefault, false);
    document.removeEventListener('drop', _dragPreventDefault, false);
  }

  _startDragging = (event: Event) => {
    event.preventDefault();
    this.setState({ isDragging: true });
  };

  _stopDragging = (event: Event) => {
    event.preventDefault();
    this.setState({ isDragging: false });
  };

  _handleProfileDrop = (event: DragEvent) => {
    event.preventDefault();
    this.setState({ isDragging: false });

    if (!event.dataTransfer) {
      return;
    }

    const { files } = event.dataTransfer;
    if (files.length > 0) {
      this.props.retrieveProfileFromFile(files[0]);
    }
  };

  render() {
    const { className, children, render } = this.props;

    const message = (
      <div
        className={classNames(
          'dragAndDropMessageWrapper',
          this.state.isDragging ? 'dragging' : false
        )}
      >
        <div className="dragAndDropMessage">Drop a saved profile here</div>
      </div>
    );

    return (
      <>
        <div
          className={classNames(className, 'dragAndDropArea')}
          onDragEnter={this._startDragging}
          onDragExit={this._stopDragging}
          onDrop={this._handleProfileDrop}
        >
          {render ? render(message) : children}
        </div>
        {/* If we weren't provided a render prop, have the message div
          as a sibling to the area div above. The area div creates its
          own stacking context, so even if it contains children with
          high z-indexes, the message div will still appear on top
          when shown.*/
        render === undefined && message}
      </>
    );
  }
}

export default DragAndDrop;
