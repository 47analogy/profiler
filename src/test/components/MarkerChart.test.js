/* This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at http://mozilla.org/MPL/2.0/. */

// @flow
import * as React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';

import MarkerChart from '../../components/marker-chart';
import { changeSelectedTab } from '../../actions/app';

import mockCanvasContext from '../fixtures/mocks/canvas-context';
import { storeWithProfile } from '../fixtures/stores';
import { getProfileWithMarkers } from '../fixtures/profiles/make-profile';
import { getBoundingBox } from '../fixtures/utils';
import mockRaf from '../fixtures/mocks/request-animation-frame';

const MARKERS = [
  ['Marker A', 0, { startTime: 0, endTime: 10 }],
  ['Marker B', 0, { startTime: 0, endTime: 10 }],
  ['Marker C', 5, { startTime: 5, endTime: 15 }],
  [
    'Very very very very very very Very very very very very very Very very very very very very Very very very very very very Very very very very very very long Marker D',
    6,
    { startTime: 5, endTime: 15 },
  ],
  ['Dot marker E', 4, { startTime: 4, endTime: 4 }],
  ['Non-interval marker F without data', 7, null],
  [
    'Marker G type DOMEvent',
    5,
    {
      type: 'tracing',
      category: 'DOMEvent',
      eventType: 'click',
      interval: 'start',
      phase: 2,
    },
  ],
  [
    'Marker G type DOMEvent',
    10,
    {
      type: 'tracing',
      category: 'DOMEvent',
      eventType: 'click',
      interval: 'end',
      phase: 2,
    },
  ],
];

const NETWORK_MARKERS = [
  [
    'Load event',
    11,
    {
      type: 'Network',
      startTime: 11,
      endTime: 12,
      id: 31666793873480,
      status: 'STATUS_START',
      pri: 0,
      URI: 'https://tiles.services.mozilla.com/v3/links/ping-centre',
    },
  ],
];

function setupWithProfile(profile) {
  const flushRafCalls = mockRaf();
  const ctx = mockCanvasContext();

  /**
   * Mock out any created refs for the components with relevant information.
   */
  function createNodeMock(element) {
    // <ChartCanvas><canvas /></ChartCanvas>
    if (element.type === 'canvas') {
      return {
        getBoundingClientRect: () => getBoundingBox(200, 300),
        getContext: () => ctx,
        style: {},
      };
    }
    // <ChartViewport />
    if (element.props.className.split(' ').includes('chartViewport')) {
      return {
        getBoundingClientRect: () => getBoundingBox(200, 300),
      };
    }
    return null;
  }

  const store = storeWithProfile(profile);
  store.dispatch(changeSelectedTab('marker-chart'));

  const markerChart = renderer.create(
    <Provider store={store}>
      <MarkerChart />
    </Provider>,
    { createNodeMock }
  );

  return {
    markerChart,
    flushRafCalls,
    store,
    flushDrawLog: () => ctx.__flushDrawLog(),
  };
}

it('renders MarkerChart correctly', () => {
  window.devicePixelRatio = 1;

  const profile = getProfileWithMarkers([...MARKERS, ...NETWORK_MARKERS]);
  const { flushRafCalls, store, markerChart, flushDrawLog } = setupWithProfile(
    profile
  );

  store.dispatch(changeSelectedTab('marker-chart'));
  flushRafCalls();

  let drawCalls = flushDrawLog();
  expect(markerChart).toMatchSnapshot();
  expect(drawCalls).toMatchSnapshot();

  store.dispatch(changeSelectedTab('network-chart'));
  flushRafCalls();

  drawCalls = flushDrawLog();
  expect(markerChart).toMatchSnapshot();
  expect(drawCalls).toMatchSnapshot();

  delete window.devicePixelRatio;
});

describe('Empty Reasons', () => {
  it('shows a reason when a profile has no marker', () => {
    const profile = getProfileWithMarkers([]);
    const { store, markerChart } = setupWithProfile(profile);

    store.dispatch(changeSelectedTab('marker-chart'));
    expect(markerChart).toMatchSnapshot();
  });

  it('shows a reason when a profil has no network markers', () => {
    const profile = getProfileWithMarkers(MARKERS);
    const { store, markerChart } = setupWithProfile(profile);

    store.dispatch(changeSelectedTab('network-chart'));
    expect(markerChart).toMatchSnapshot();
  });
});
