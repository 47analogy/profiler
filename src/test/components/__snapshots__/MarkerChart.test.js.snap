// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MarkerChart EmptyReasons shows a reason when a profile has no markers 1`] = `
<div
  class="EmptyReasons"
>
  <h3>
    The 
    marker chart
     is empty for “
    Empty
    ”
  </h3>
  <p>
    This thread contains no markers for this chart.
  </p>
</div>
`;

exports[`MarkerChart EmptyReasons shows a reason when a profile's markers have been filtered out 1`] = `
<div
  class="EmptyReasons"
>
  <h3>
    The 
    marker chart
     is empty for “
    Empty
    ”
  </h3>
  <p>
    All markers were filtered out by the current selection or search term.
  </p>
</div>
`;

exports[`MarkerChart renders the hoveredItem markers properly 1`] = `
Array [
  Array [
    "set fillStyle",
    "#ffffff",
  ],
  Array [
    "fillRect",
    0,
    0,
    365,
    300,
  ],
  Array [
    "set lineWidth",
    1,
  ],
  Array [
    "set fillStyle",
    "#45a1ff",
  ],
  Array [
    "fillRect",
    151,
    1,
    123,
    1,
  ],
  Array [
    "fillRect",
    150,
    2,
    125,
    11,
  ],
  Array [
    "fillRect",
    151,
    13,
    123,
    1,
  ],
  Array [
    "measureText",
    "",
  ],
  Array [
    "set fillStyle",
    "#45a1ff",
  ],
  Array [
    "fillRect",
    289,
    1,
    48,
    1,
  ],
  Array [
    "fillRect",
    288,
    2,
    50,
    11,
  ],
  Array [
    "fillRect",
    289,
    13,
    48,
    1,
  ],
  Array [
    "measureText",
    "",
  ],
  Array [
    "set fillStyle",
    "#45a1ff",
  ],
  Array [
    "fillRect",
    151,
    17,
    123,
    1,
  ],
  Array [
    "fillRect",
    150,
    18,
    125,
    11,
  ],
  Array [
    "fillRect",
    151,
    29,
    123,
    1,
  ],
  Array [
    "measureText",
    "",
  ],
  Array [
    "set fillStyle",
    "#45a1ff",
  ],
  Array [
    "fillRect",
    151,
    33,
    35.5,
    1,
  ],
  Array [
    "fillRect",
    150,
    34,
    37.5,
    11,
  ],
  Array [
    "fillRect",
    151,
    45,
    35.5,
    1,
  ],
  Array [
    "measureText",
    "",
  ],
  Array [
    "set fillStyle",
    "#45a1ff",
  ],
  Array [
    "beginPath",
  ],
  Array [
    "arc",
    205,
    71.5,
    3.75,
    0,
    6.283185307179586,
  ],
  Array [
    "fill",
  ],
  Array [
    "set fillStyle",
    "#45a1ff",
  ],
  Array [
    "fillRect",
    214,
    81,
    123,
    1,
  ],
  Array [
    "fillRect",
    213,
    82,
    125,
    11,
  ],
  Array [
    "fillRect",
    214,
    93,
    123,
    1,
  ],
  Array [
    "measureText",
    "",
  ],
  Array [
    "set fillStyle",
    "#45a1ff",
  ],
  Array [
    "fillRect",
    214,
    97,
    60.5,
    1,
  ],
  Array [
    "fillRect",
    213,
    98,
    62.5,
    11,
  ],
  Array [
    "fillRect",
    214,
    109,
    60.5,
    1,
  ],
  Array [
    "set fillStyle",
    "white",
  ],
  Array [
    "fillText",
    "click",
    216,
    107,
  ],
  Array [
    "set fillStyle",
    "#45a1ff",
  ],
  Array [
    "beginPath",
  ],
  Array [
    "arc",
    243,
    119.5,
    3.75,
    0,
    6.283185307179586,
  ],
  Array [
    "fill",
  ],
  Array [
    "set fillStyle",
    "#45a1ff",
  ],
  Array [
    "fillRect",
    264,
    129,
    85.5,
    1,
  ],
  Array [
    "fillRect",
    263,
    130,
    87.5,
    11,
  ],
  Array [
    "fillRect",
    264,
    141,
    85.5,
    1,
  ],
  Array [
    "measureText",
    "",
  ],
  Array [
    "set fillStyle",
    "Highlight",
  ],
  Array [
    "fillRect",
    176,
    49,
    73,
    1,
  ],
  Array [
    "fillRect",
    175,
    50,
    75,
    11,
  ],
  Array [
    "fillRect",
    176,
    61,
    73,
    1,
  ],
  Array [
    "set fillStyle",
    "HighlightText",
  ],
  Array [
    "fillText",
    "Marker B",
    178,
    59,
  ],
  Array [
    "set fillStyle",
    "#ededf0",
  ],
  Array [
    "fillRect",
    149,
    0,
    1,
    300,
  ],
  Array [
    "fillRect",
    0,
    15,
    365,
    1,
  ],
  Array [
    "fillRect",
    0,
    31,
    365,
    1,
  ],
  Array [
    "fillRect",
    0,
    47,
    365,
    1,
  ],
  Array [
    "fillRect",
    0,
    63,
    365,
    1,
  ],
  Array [
    "fillRect",
    0,
    79,
    365,
    1,
  ],
  Array [
    "fillRect",
    0,
    95,
    365,
    1,
  ],
  Array [
    "fillRect",
    0,
    111,
    365,
    1,
  ],
  Array [
    "fillRect",
    0,
    127,
    365,
    1,
  ],
  Array [
    "fillRect",
    0,
    143,
    365,
    1,
  ],
  Array [
    "set fillStyle",
    "#000000",
  ],
  Array [
    "fillText",
    "Marker A",
    5,
    11,
  ],
  Array [
    "fillText",
    "Marker H with no start",
    5,
    43,
  ],
  Array [
    "fillText",
    "UserTiming",
    5,
    59,
  ],
  Array [
    "fillText",
    "Dot marker E",
    5,
    75,
  ],
  Array [
    "fillText",
    "Very very very very very ver…",
    5,
    91,
  ],
  Array [
    "fillText",
    "Marker G type DOMEvent",
    5,
    107,
  ],
  Array [
    "fillText",
    "Non-interval marker F withou…",
    5,
    123,
  ],
  Array [
    "fillText",
    "Marker H with no end",
    5,
    139,
  ],
]
`;

exports[`MarkerChart renders the hoveredItem markers properly 2`] = `
<div
  class="tooltip"
  data-testid="tooltip"
  style="left: 189px; top: 70px;"
>
  <div
    class="tooltipMarker"
  >
    <div
      class="tooltipHeader"
    >
      <div
        class="tooltipOneLine"
      >
        <div
          class="tooltipTiming"
        >
          6.0ms
        </div>
        <div
          class="tooltipTitle"
        >
          UserTiming
        </div>
      </div>
      <div
        class="tooltipDetails"
      >
        <div
          class="tooltipLabel"
        >
          Thread
          :
        </div>
        Empty
      </div>
    </div>
    <div
      class="tooltipDetails"
    >
      <div
        class="tooltipLabel"
      >
        Name
        :
      </div>
      Marker B
    </div>
  </div>
</div>
`;

exports[`MarkerChart renders the normal marker chart and matches the snapshot 1`] = `
<div
  aria-labelledby="marker-chart-tab-button"
  class="markerChart"
  id="marker-chart-tab"
  role="tabpanel"
>
  <div
    class="markerSettings"
  >
    <div
      class="markerSettingsSpacer"
    />
    <div
      class="panelSearchField markerSettingsSearchField"
    >
      <label
        class="panelSearchFieldLabel"
      >
        Filter Markers: 
        <form
          class="idleSearchField panelSearchFieldInput"
        >
          <input
            class="idleSearchFieldInput photon-input"
            name="search"
            placeholder="Enter filter terms"
            required=""
            title="Only display markers that match a certain name"
            type="search"
            value=""
          />
          <input
            class="idleSearchFieldButton"
            tabindex="-1"
            type="reset"
          />
        </form>
        <div
          class="panelSearchFieldIntroduction isHidden"
        >
          Did you know you can use the comma (,) to search using several terms?
        </div>
      </label>
    </div>
  </div>
  <div
    class="react-contextmenu-wrapper treeViewContextMenu"
  >
    <div
      class="chartViewport"
      tabindex="0"
    >
      <div>
        <canvas
          class="chartCanvas markerChartCanvas"
          height="600"
          style="width: 365px; height: 300px;"
          width="730"
        />
      </div>
      <div
        class="chartViewportScroll hidden"
      >
        Zoom Chart:
        <kbd
          class="chartViewportScrollKbd"
        >
          Ctrl + Scroll
        </kbd>
        or
        <kbd
          class="chartViewportScrollKbd"
        >
          Shift + Scroll
        </kbd>
      </div>
    </div>
  </div>
</div>
`;

exports[`MarkerChart renders the normal marker chart and matches the snapshot 2`] = `
Array [
  Array [
    "scale",
    1,
    1,
  ],
  Array [
    "scale",
    2,
    2,
  ],
  Array [
    "set fillStyle",
    "#ffffff",
  ],
  Array [
    "fillRect",
    0,
    0,
    365,
    300,
  ],
  Array [
    "set lineWidth",
    1,
  ],
  Array [
    "set fillStyle",
    "#45a1ff",
  ],
  Array [
    "measureText",
    "abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ.()< /:-_",
  ],
  Array [
    "measureText",
    "…",
  ],
  Array [
    "fillRect",
    151,
    1,
    123,
    1,
  ],
  Array [
    "fillRect",
    150,
    2,
    125,
    11,
  ],
  Array [
    "fillRect",
    151,
    13,
    123,
    1,
  ],
  Array [
    "measureText",
    "",
  ],
  Array [
    "set fillStyle",
    "#45a1ff",
  ],
  Array [
    "fillRect",
    288.5,
    1,
    48,
    1,
  ],
  Array [
    "fillRect",
    287.5,
    2,
    50,
    11,
  ],
  Array [
    "fillRect",
    288.5,
    13,
    48,
    1,
  ],
  Array [
    "measureText",
    "",
  ],
  Array [
    "set fillStyle",
    "#45a1ff",
  ],
  Array [
    "fillRect",
    151,
    17,
    123,
    1,
  ],
  Array [
    "fillRect",
    150,
    18,
    125,
    11,
  ],
  Array [
    "fillRect",
    151,
    29,
    123,
    1,
  ],
  Array [
    "measureText",
    "",
  ],
  Array [
    "set fillStyle",
    "#45a1ff",
  ],
  Array [
    "fillRect",
    151,
    33,
    35.5,
    1,
  ],
  Array [
    "fillRect",
    150,
    34,
    37.5,
    11,
  ],
  Array [
    "fillRect",
    151,
    45,
    35.5,
    1,
  ],
  Array [
    "measureText",
    "",
  ],
  Array [
    "set fillStyle",
    "#45a1ff",
  ],
  Array [
    "fillRect",
    176,
    49,
    73,
    1,
  ],
  Array [
    "fillRect",
    175,
    50,
    75,
    11,
  ],
  Array [
    "fillRect",
    176,
    61,
    73,
    1,
  ],
  Array [
    "measureText",
    "Marker B",
  ],
  Array [
    "set fillStyle",
    "white",
  ],
  Array [
    "fillText",
    "Marker B",
    178,
    59,
  ],
  Array [
    "set fillStyle",
    "#45a1ff",
  ],
  Array [
    "beginPath",
  ],
  Array [
    "arc",
    205,
    71.5,
    3.75,
    0,
    6.283185307179586,
  ],
  Array [
    "fill",
  ],
  Array [
    "set fillStyle",
    "#45a1ff",
  ],
  Array [
    "fillRect",
    213.5,
    81,
    123,
    1,
  ],
  Array [
    "fillRect",
    212.5,
    82,
    125,
    11,
  ],
  Array [
    "fillRect",
    213.5,
    93,
    123,
    1,
  ],
  Array [
    "measureText",
    "",
  ],
  Array [
    "set fillStyle",
    "#45a1ff",
  ],
  Array [
    "fillRect",
    213.5,
    97,
    60.5,
    1,
  ],
  Array [
    "fillRect",
    212.5,
    98,
    62.5,
    11,
  ],
  Array [
    "fillRect",
    213.5,
    109,
    60.5,
    1,
  ],
  Array [
    "measureText",
    "click",
  ],
  Array [
    "set fillStyle",
    "white",
  ],
  Array [
    "fillText",
    "click",
    215.5,
    107,
  ],
  Array [
    "set fillStyle",
    "#45a1ff",
  ],
  Array [
    "beginPath",
  ],
  Array [
    "arc",
    242.5,
    119.5,
    3.75,
    0,
    6.283185307179586,
  ],
  Array [
    "fill",
  ],
  Array [
    "set fillStyle",
    "#45a1ff",
  ],
  Array [
    "fillRect",
    263.5,
    129,
    85.5,
    1,
  ],
  Array [
    "fillRect",
    262.5,
    130,
    87.5,
    11,
  ],
  Array [
    "fillRect",
    263.5,
    141,
    85.5,
    1,
  ],
  Array [
    "measureText",
    "",
  ],
  Array [
    "set fillStyle",
    "#ededf0",
  ],
  Array [
    "fillRect",
    149,
    0,
    1,
    300,
  ],
  Array [
    "fillRect",
    0,
    15,
    365,
    1,
  ],
  Array [
    "fillRect",
    0,
    31,
    365,
    1,
  ],
  Array [
    "fillRect",
    0,
    47,
    365,
    1,
  ],
  Array [
    "fillRect",
    0,
    63,
    365,
    1,
  ],
  Array [
    "fillRect",
    0,
    79,
    365,
    1,
  ],
  Array [
    "fillRect",
    0,
    95,
    365,
    1,
  ],
  Array [
    "fillRect",
    0,
    111,
    365,
    1,
  ],
  Array [
    "fillRect",
    0,
    127,
    365,
    1,
  ],
  Array [
    "fillRect",
    0,
    143,
    365,
    1,
  ],
  Array [
    "set fillStyle",
    "#000000",
  ],
  Array [
    "measureText",
    "Marker A",
  ],
  Array [
    "fillText",
    "Marker A",
    5,
    11,
  ],
  Array [
    "measureText",
    "Marker H with no start",
  ],
  Array [
    "fillText",
    "Marker H with no start",
    5,
    43,
  ],
  Array [
    "measureText",
    "UserTiming",
  ],
  Array [
    "fillText",
    "UserTiming",
    5,
    59,
  ],
  Array [
    "measureText",
    "Dot marker E",
  ],
  Array [
    "fillText",
    "Dot marker E",
    5,
    75,
  ],
  Array [
    "measureText",
    "Very very very very very very Very very very very very very Very very very very very very Very very very very very very Very very very very very very long Marker D",
  ],
  Array [
    "fillText",
    "Very very very very very ver…",
    5,
    91,
  ],
  Array [
    "measureText",
    "Marker G type DOMEvent",
  ],
  Array [
    "fillText",
    "Marker G type DOMEvent",
    5,
    107,
  ],
  Array [
    "measureText",
    "Non-interval marker F without data",
  ],
  Array [
    "fillText",
    "Non-interval marker F withou…",
    5,
    123,
  ],
  Array [
    "measureText",
    "Marker H with no end",
  ],
  Array [
    "fillText",
    "Marker H with no end",
    5,
    139,
  ],
]
`;
